gpumon/gpumon.ml

```diff
@@ -304,11 +330,13 @@ let () =
     Process.D.info "Received signal %d: deregistering plugin %s..." signal plugin_name;
     exit 0
   in
-  (* gpumon idl server *)
   let module Gpumon_server = Gpumon_server.Make(struct
       let interface = maybe_interface
     end) in
-  let module Server = Gpumon_interface.RPC_API(Idl.GenServerExn ()) in
+  (* create daemon module to bind server call declarations to implementations *)
+  let module Daemon = Make(Gpumon_server.Nvidia) in
```

Comment:
Should we pass Gpumon_server instead and change the functor parameter signature?